crate/pyo3/src/py_kms_client.rs

```diff
@@ -24,6 +27,47 @@ use rustls::Certificate;
 
 use crate::py_kms_object::KmsObject;
 
+/// Create a Rekey Keypair request from PyO3 arguments
+/// Returns a PyO3 Future
+macro_rules! rekey_keypair {
+    (
+        $self:ident,
+        $attributes:expr,
+        $master_secret_key_identifier:expr,
+        $tags:expr,
+        $action:expr,
+        $py:ident
+    ) => {{
+        let id = match ($master_secret_key_identifier, $tags) {
+            (Some(key_id), _) => key_id,
+            (_, Some(tags)) => serde_json::to_string(&tags)
+                .map_err(|_e| PyException::new_err("invalid tag(s) specified"))?,
+            _ => return Err(PyException::new_err("please specify a key id or tags")),
+        };
```

Comment:
Would it be a good idea to create a `Handle` type to avoid the `_` case?
```rust
pub enum KeyHandle {
    Id(String),
    Tags(Vec<Tag>),
}

\\ then...
let id = match handle {
    KeyHand::Id(id) => id,
    KeyHandle::Tags(tags) => serde_json::to_string(&tags).map_err(...))?,
};
```

You could even implement this as a method like `KeyHandle::try_to_string()`.